### PR TITLE
Improve Validators

### LINKED ISSUE #76: Increase validators info
More info https://developer.vocdoni.io/vocdoni-api/list-validators

The validator list should show `validatorAddress` instead of `address` as a main address for the validator. Then on the blocks section, make a link of the validator address to its corresponding validatorAddress `href /validators#validatorAddress`.

Depends on https://github.com/vocdoni/vocdoni-sdk/pull/402

### PR DESCRIPTION
**Pull request by selankon**
Fix https://github.com/vocdoni/explorer/issues/76
Depends on https://github.com/vocdoni/vocdoni-sdk/pull/402/files

- [x] Implement validators detail page
- [x] Fix block link to validator

It extracts a new component called `IconLink` used to create links with icon. The idea is to implement some kind of links on different places with a leading icon like: 

image

This is related to https://github.com/vocdoni/explorer/issues/79 which will need this component

### CODE REVIEW

File: src/components/Validators/ValidatorCard.tsx [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
-          <Text fontWeight='bold' wordBreak='break-all'>
-            {ensure0x(validator.address)}
-          </Text>
-          <Flex fontSize={'sm'} direction={'column'} gap={2}>
-            <HStack gap={1}>
+      <Link as={RouterLink} to={generatePath(RoutePath.Validator, { address: validator.validatorAddress })}>
+        <CardBody>
+          <Flex gap={2} direction={'column'}>
+            <Text fontWeight='bold' wordBreak='break-all'>
+              {validator.name && `(${validator.name})`} {ensure0x(validator.validatorAddress)}
```
Comment (elboletaire) [inline src/components/Validators/ValidatorCard.tsx]: Create a component to better handle this, and show it like:

With unknown name: `just the sha`
With known name: `Name (sha)`

Idealy, the `(sha)` should be shown in a different style (smaller size and lighter color maybe [?])

File: src/components/Validators/Detail.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            fontSize={'xl'}
+            p={0}
+          >
+            {validatorAddress}
+          </ReducedTextAndCopy>
+          {!!name && <Text color={'lighterText'}>({name})</Text>}
+        </Flex>
+        <HStack color={'lighterText'} fontWeight={'bold'}>
+          <Text>
+            <Trans i18nKey={'validators.validator_details'}>Joint on block</Trans>
```
Comment (elboletaire) [inline src/components/Validators/Detail.tsx]: Joined**

**Review by elboletaire — CHANGES_REQUESTED** [review]

File: src/components/Validators/ValidatorCard.tsx (hunk 1 above)
Comment (selankon) [reply src/components/Validators/ValidatorCard.tsx]: efcf103

This commit makes an extrapolation of the component, this is the result

image

image

image

image

File: src/components/Validators/Detail.tsx (hunk 2 above)
Comment (selankon) [reply src/components/Validators/Detail.tsx]: dfc78c3

**Comment by selankon** [comment]
@elboletaire I added a link to the validator account address, I don't know if it has sense to have this info available...

92589d2

**Comment by elboletaire** [comment]
> @elboletaire I added a link to the validator account address, I don't know if it has sense to have this info available...
> 
> [92589d2](https://github.com/vocdoni/explorer/commit/92589d2500aab9ba1a04839d104b226dc3d36aa8)

I feel like there are too much copy options in the validators list now. I'd leave them without the copies I think, since we now have a view and you can see all the details there, it's better to keep the copy buttons there only.

Well I see it improves with names defined, but it still feels weird when they aren't:
imatge
imatge

**Comment by selankon** [comment]
> > @elboletaire I added a link to the validator account address, I don't know if it has sense to have this info available...
> > [92589d2](https://github.com/vocdoni/explorer/commit/92589d2500aab9ba1a04839d104b226dc3d36aa8)
> 
> I feel like there are too much copy options in the validators list now. I'd leave them without the copies I think, since we now have a view and you can see all the details there, it's better to keep the copy buttons there only.

This approach is more on the way you propose 7a18c93

**Review by elboletaire — APPROVED** [review]

**Pull request merged**